### ISSUE TITLE
connector fails to build against CUPS <1.7

### ISSUE THREAD

**Issue by Jibber81** [google/cups-connector#31]
go get github.com/google/cups-connector/connector
# github.com/google/cups-connector/cups
could not determine kind of name for C.HTTP_ENCRYPTION_ALWAYS
could not determine kind of name for C.HTTP_ENCRYPTION_IF_REQUESTED
could not determine kind of name for C.HTTP_ENCRYPTION_NEVER
could not determine kind of name for C.HTTP_ENCRYPTION_REQUIRED
could not determine kind of name for C.HTTP_STATUS_NOT_MODIFIED
could not determine kind of name for C.HTTP_STATUS_OK
could not determine kind of name for C.IPP_OP_CUPS_GET_PRINTERS
could not determine kind of name for C.IPP_OP_GET_JOB_ATTRIBUTES
could not determine kind of name for C.IPP_STATUS_ERROR_NOT_FOUND
could not determine kind of name for C.IPP_STATUS_OK
could not determine kind of name for C.httpConnect2
could not determine kind of name for C.ippGetStatusCode
could not determine kind of name for C.ippValidateAttributes

**Comment by jacobmarble** [comment]
Hi, thanks for giving this a try. It looks those are defined in cups/http.h and cups/ipp.h, which are not included in cups-connector/cups/cups.h

What operating system (distro, version) and what version of CUPS (if not the packaged one)?

**Comment by Jibber81** [comment]
I try to install it on a raspberry pi. 

OS: Debian GNU/Linux 7.8 (wheezy)
Cups version: 1.5.3

**Comment by jacobmarble** [comment]
Currently, the project does not support CUPS < 1.6 but let me look into fixing that, and I'll update you later.

**Comment by jacobmarble** [comment]
My Raspberry Pi is running Debian 8.0, which comes with CUPS 1.7.5 and explains why I haven't seen this problem. Is switching distros an option for you?

**Comment by Jibber81** [comment]
Thanks for your answer. I was able to upgrade my Pi to "jessie" although I am a linux noob. Now I am able to start the cups-connector and I just printed the first page, yeah :-)

How can I start the cups-connector automatically when I reboot the system? Do you have a sample from your Pi installation? 

Thanks a lot and great job!

**Comment by jacobmarble** [comment]
Edit /etc/rc.local and add this before ```exit 0```:
```
# CUPS Connector:
#   su ... pi: run as user "pi"
#   --login: environment similar to "pi" instead of "root"
#   --command: run this thing
#   &: run the command "in the background"
su --login --command "go/bin/connector" pi &
```

The directory ```/var/run/cups-connector/monitor.sock``` will not exist on boot, and the process won't start. The easiest way to fix this is to edit cups-connector.config.json and change ```monitor_socket_filename``` to something like ```/tmp/cups-connector-monitor.sock```

Let me know how that works for you, and I'll update the README for the next Raspberry Pi user.

**Comment by Jibber81** [comment]
Yes, this seems to work for me. Thanks a lot for your support!

**Comment by Pilot001** [comment]
I do have the same problem (as mentioned in the subject line). I'm running openSUSE and it comes with cups 1.5.4, even in the latest distro version:
# github.com/google/cups-connector/cups
could not determine kind of name for C.HTTP_ENCRYPTION_ALWAYS
could not determine kind of name for C.HTTP_ENCRYPTION_IF_REQUESTED
could not determine kind of name for C.HTTP_ENCRYPTION_NEVER
could not determine kind of name for C.HTTP_ENCRYPTION_REQUIRED
could not determine kind of name for C.HTTP_STATUS_NOT_MODIFIED
could not determine kind of name for C.HTTP_STATUS_OK
could not determine kind of name for C.IPP_OP_CUPS_GET_PRINTERS
could not determine kind of name for C.IPP_OP_GET_JOB_ATTRIBUTES
could not determine kind of name for C.IPP_STATUS_ERROR_NOT_FOUND
could not determine kind of name for C.IPP_STATUS_OK
could not determine kind of name for C.httpConnect2
could not determine kind of name for C.ippGetStatusCode
could not determine kind of name for C.ippValidateAttributes

**Comment by jacobmarble** [comment]
**If you are reading this and your distribution comes with CUPS <1.7, please add a comment.**

Found this note about openSUSE and printing. Looks like the next release should get a CUPS update:
https://en.opensuse.org/Portal:Printing

I need to look at the CUPS client code to verify that this isn't actually a big change to make. Of the lines quoted above, the top 10 are constant values with older names available, the 11th should be replaceable with the older httpConnect() function, and the last two I need to at closer.

However, this is currently low priority relative to other issues because:
- only one distro needs the downgrade so far (see bold text above)
- a new release of openSUSE should arrive soon

**Comment by mceachen** [comment]
(well, I'd include Rasbian, too, so that's two distros using <v1.7 CUPS)

**Comment by jacobmarble** [comment]
RHEL and CentOS are packaged with CUPS 1.6.3.

I have a patch ready, and will test it on Raspbian early next week.

**Comment by Maxr1998** [comment]
I am using a Raspberry Pi with the slimmed down DietPI os, which is based on Raspbian Wheezy.
Upgrading to jessie resulted two times in bricking the system, so I'd need support for 1.5.3..

**Comment by Maxr1998** [comment]
I just tested the current version, and it threw the same errors as OP..

**Comment by jacobmarble** [comment]
This *should* fix everyone back to at least CUPS 1.5.

**Comment by jacobmarble** [comment]
Here are some [step-by-step instructions](https://github.com/google/cups-connector/wiki/Raspberry-Pi-step-by-step) to installing on Raspberry Pi.
